be/src/storage/compaction_scheduler.cpp

```diff
@@ -206,45 +199,42 @@ std::shared_ptr<CompactionTask> CompactionScheduler::_try_get_next_compaction_ta
     // tmp_tablets save the tmp picked candidates tablets
     std::vector<CompactionCandidate> tmp_candidates;
     CompactionCandidate compaction_candidate;
-    bool found = false;
     std::shared_ptr<CompactionTask> compaction_task;
+
     while (true) {
         if (!_can_schedule_next()) {
             VLOG(2) << "_can_schedule_next is false. skip";
-            break;
+            return nullptr;
```

Comment:
should not return directly since `tmp_candidates` haven't insert to compaction_manager

Reply:
`tmp_candidates` is empty here, so it's ok not to insert the `tmp_candidates` to compaction manager.

Reply:
it's a while loop, tmp_candidates may be inserted before this round?

Reply:
Updated.